plugins/Microsoft.DocAsCode.Build.UniversalReference/ModelConverter/ModelConverter.cs

```diff
@@ -0,0 +1,278 @@
+                    };
+                }).ToList();
+        }
+
+        public static List<ApiLanguageValuePair<List<ApiInheritanceTreeBuildOutput>>> ToApiListInDevlangsResolvingApiNames(List<InheritanceTree> defaultValue, SortedList<string, List<InheritanceTree>> values, string[] supportedLanguages, IReadOnlyDictionary<string, ApiNames> references)
+        {
+            if (defaultValue == null || supportedLanguages == null || supportedLanguages.Length == 0 || references == null)
+            {
+                return null;
+            }
+            return ToApiListInDevLangs(defaultValue, values, supportedLanguages)
+                .Select(pair =>
+                {
+                    return new ApiLanguageValuePair<List<ApiInheritanceTreeBuildOutput>>
+                    {
+                        Language = pair.Language,
+                        Value = pair.Value.Select(item => ResolveInheritanceTree(item, supportedLanguages, references)).ToList()
+                    };
+                }).ToList();
+        }
+
+        public static List<ApiLanguageValuePair<List<ApiExceptionInfoBuildOutput>>> ToApiListInDevlangsResolvingApiNames(List<ExceptionInfo> defaultValue, SortedList<string, List<ExceptionInfo>> values, string[] supportedLanguages, IReadOnlyDictionary<string, ApiNames> references)
+        {
+            if (defaultValue == null || supportedLanguages == null || supportedLanguages.Length == 0 || references == null)
+            {
+                return null;
+            }
+            return ToApiListInDevLangs(defaultValue, values, supportedLanguages)
+                .Select(pair =>
+                {
+                    return new ApiLanguageValuePair<List<ApiExceptionInfoBuildOutput>>
+                    {
+                        Language = pair.Language,
+                        Value = pair.Value.Select(item =>
+                        {
+                            return new ApiExceptionInfoBuildOutput
+                            {
+                                Type = ResolveApiNames(item.Type, supportedLanguages, references),
+                                Description = item.Description,
+                                Metadata = item.Metadata
+                            };
+                        }).ToList()
+                    };
+                }).ToList();
+        }
+
+        public static List<ApiLanguageValuePair<ApiParameterBuildOutput>> ToApiListInDevLangsResolvingApiNames(ApiParameter defaultValue, SortedList<string, ApiParameter> values, string[] supportedLanguages, IReadOnlyDictionary<string, ApiNames> references)
+        {
+            if (defaultValue == null || supportedLanguages == null || supportedLanguages.Length == 0 || references == null)
+            {
+                return null;
+            }
+            return (List<ApiLanguageValuePair<ApiParameterBuildOutput>>)ToApiListInDevLangs(defaultValue, values, supportedLanguages)
```

Comment:
can remove (List<ApiLanguageValuePair<ApiParameterBuildOutput>>)

---
In reply to: [114265699](https://github.com/dotnet/docfx/pull/1618#discussion_r114265699) [](ancestors = 114265699)